src/pudl/extract/xbrl.py

```diff
@@ -0,0 +1,166 @@
+                    InstanceBuilder(
+                        io.BytesIO(archive.open(latest_filing).read()), filing_id
+                    )
+                )
+
+        return filings
+
+
+def _get_sqlite_engine(
+    form_number: int, pudl_settings: dict, clobber: bool
+) -> sa.engine.Engine:
+    """Create SQLite engine for specified form and drop tables.
+
+    Args:
+        form_number: FERC form number.
+        pudl_settings: Dictionary of settings.
+        clobber: Flag indicating whether or not to drop tables.
+    """
+    # Read in the structure of the DB, if it exists
+    logger.info(f"Dropping the old FERC Form {form_number} SQLite DB if it exists.")
+    sqlite_engine = sa.create_engine(pudl_settings[f"ferc{form_number}_xbrl_db"])
+    try:
+        # So that we can wipe it out
+        pudl.helpers.drop_tables(sqlite_engine, clobber=clobber)
+    except sa.exc.OperationalError:
+        pass
+
+    return sqlite_engine
+
+
+def xbrl2sqlite(
+    ferc_to_sqlite_settings: FercToSqliteSettings = FercToSqliteSettings(),
+    pudl_settings: dict = None,
+    clobber: bool = False,
+    datastore: Datastore = None,
+    batch_size: int | None = None,
+    workers: int | None = None,
+):
+    """Clone the FERC Form 1 XBRL Databsae to SQLite.
+
+    Args:
+        ferc_to_sqlite_settings: Object containing Ferc to SQLite validated
+            settings.
+        pudl_settings: Dictionary containing paths and database URLs
+            used by PUDL.
+        clobber: Flag indicating whether or not to drop tables.
+        datastore: Instance of a datastore to access the resources.
+        batch_size: Number of XBRL filings to process in a single CPU process.
+        workers: Number of CPU processes to create for processing XBRL filings.
+
+    Returns:
+        None
+
+    """
+    datastore = FercXbrlDatastore(datastore)
+
+    # Loop through all other forms and perform conversion
+    for form in XBRL_FORM_NUMBERS:
+        # Get desired settings object
+        settings = ferc_to_sqlite_settings.get_xbrl_dataset_settings(form)
+
+        # If no settings for form in question, skip
+        if settings is None:
+            continue
+
+        sqlite_engine = _get_sqlite_engine(form, pudl_settings, clobber)
+
+        convert_form(
+            settings,
+            form,
+            datastore,
+            sqlite_engine,
+            pudl_settings=pudl_settings,
+            batch_size=batch_size,
+            workers=workers,
+        )
+
+
+def convert_form(
+    form_settings: FercGenericXbrlToSqliteSettings,
+    form_number: int,
+    datastore: FercXbrlDatastore,
+    sqlite_engine: sa.engine.Engine,
+    pudl_settings: dict = None,
+    batch_size: int | None = None,
+    workers: int | None = None,
+):
+    """Clone a single FERC XBRL form to SQLite.
+
+    Args:
+        form_settings: Validated settings for converting the desired XBRL form to SQLite.
+        form_number: FERC form number.
```

Comment:
Indication of what the allowable form numbers are here would be helpful.

Reply:
Do you think making this an enum is sufficient?